packages/harmony/src/components/input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,213 @@
+import { forwardRef, type ComponentPropsWithoutRef } from 'react'
+
+import cn from 'classnames'
+
+import layoutStyles from 'components/layout/layout.module.css'
+import { IconComponent, Text } from 'components/typography'
+
+import { HelperText } from './HelperText'
+import styles from './TextInput.module.css'
+import { useFocusState } from './useFocusState'
+
+export enum TextInputSize {
+  SMALL = 'small',
+  DEFAULT = 'default'
+}
+
+export type TextInputProps = Omit<
+  ComponentPropsWithoutRef<'input'>,
+  // Omitting required purely for storybook docs
+  'size' | 'required'
+> & {
+  /**
+   * Input sizes. NOTE: small inputs will not show the label
+   * @default default
+   */
+  size?: TextInputSize
+  /**
+   * Show a "x/Max" text displaying number of characters allowed
+   */
+  showMaxLength?: boolean
+  /**
+   * Toggle warning state
+   */
+  warning?: boolean
+  /**
+   * Toggle error state
+   */
+  error?: boolean
+  /**
+   * Hides the label. If the label is hidden the placeholder will show instead.
+   * @default false
+   */
+  hideLabel?: boolean
```

Comment:
Previously this was being used as the "variant", but the only behavior between the two was label or no label so I decided to make a toggle option instead